lib/internal/modules/cjs/loader.js

```diff
@@ -1277,9 +1288,37 @@ function loadESMFromCJS(mod, filename) {
   const source = getMaybeCachedSource(mod, filename);
   const cascadedLoader = require('internal/modules/esm/loader').getOrInitializeCascadedLoader();
   const isMain = mod[kIsMainSymbol];
-  // TODO(joyeecheung): we may want to invent optional special handling for default exports here.
-  // For now, it's good enough to be identical to what `import()` returns.
-  mod.exports = cascadedLoader.importSyncForRequire(filename, source, isMain);
+  const namespace = cascadedLoader.importSyncForRequire(filename, source, isMain);
+  // Tooling in the ecosystem have been using the __esModule property to recognize
+  // transpiled ESM in consuming code. For example, a 'log' package written in ESM:
+  //
+  // export function log(val) { console.log(val); }
```

Comment:
I assume this is a default export reading the following transpiled example.

```suggestion
  // export default function log(val) { console.log(val); }
```